core/keyserver/readtoken.proto

```diff
@@ -18,10 +18,25 @@ package google.keytransparency.v1;
 
 option go_package = "github.com/google/keytransparency/core/keyserver/readtoken_go_proto";
 
+import "v1/keytransparency.proto";
+
 // ReadToken can be serialized and handed to users for pagination.
 message ReadToken {
   // shard_id identifies the source for reading.
   int64 shard_id = 1;
   // low_watemark identifies the lowest (exclusive) row to return.
   int64 low_watermark = 2;
 }
+
+// ListUserRevisions token can be serialized and handed to users for pagination
+// when listing revisions.
+message ListUserRevisionsToken {
+  // request is the query being paginated over, used for validation of
+  // subsequent requests. Fields that are allowed to change between requests
+  // (such as page_token or last_verified_tree_size) will not be validated and
+  // should be omitted for brevity.
+  ListUserRevisionsRequest request = 1;
+  // revisions_returned is a running tally of the number of revisions that have
+  // been returned across paginated requests in this query.
+  int64 revisions_returned = 2;
```

Comment:
Is `revisions_returned` more like `offset` or a statistic?

I found this article to be a nice review of various paging techniques and their tradeoffs: 
https://slack.engineering/evolving-api-pagination-at-slack-1c1f644f8e12

For our use case we can efficiently add `start` + `revisions_returned` to get our read cursor, but from a design perspective, we might want to train ourselves to use read cursors to start with.

Reply:
I was thinking `revisions_returned` would be used like an `offset` in practice.

Would the read cursor be passed through to Trillian? If so, it seems like we'd need to implement batched+paged reads from Trillian first, which is an area I know nothing about yet.

Reply:
At the moment, Trillian only offers [`GetLeavesByRevision`](https://github.com/google/trillian/blob/master/trillian_map_api.proto#L125) so (for the moment), the keyserver is generating individual requests for each revision and passing them to Trillian. 

When Trillian implements `GetLeavesByRevisionRange` we can transition / add an opaque paging token to this proto that we can pass through. 

`revisions_returned` makes sense for now. The client logic should be exactly the same in both cases since this whole proto will be an opaque token.